samcli/commands/local/lib/local_lambda.py

```diff
@@ -280,7 +280,7 @@ def _make_env_vars(self, function: Function) -> EnvironmentVariables:
             aws_creds=aws_creds,
         )  # EnvironmentVariables is not yet annotated with type hints, disable mypy check for now. type: ignore
 
-    def _get_session_creds(self) -> Credentials:
+    def _get_session_creds(self) -> Optional[Credentials]:
```

Comment:
This function can potentially return `None`. Should be safe for the outside code, only typing was off.